chainer/variable.py

```diff
@@ -925,6 +925,10 @@ def _set_chainerx_grad(self, g):
                 raise RuntimeError(
                     'Cannot set a gradient to an empty variable')
         elif arr.is_backprop_required():
+            # If g is grad-stopped, require grad on it.
+            # Make a view in order not to affect the input.
+            if g is not None and not g.is_backprop_required():
+                g = g.view().require_grad()
```

Comment:
`_set_chainerx_grad` is called from the `grad_var` setter too. If the argument variable to a call to it holds a `chainerx.ndarray` which doesn't require any gradients, it will be modified to require gradients here. Maybe pass a third flag argument to this method to control it?

```python
import chainerx as chx
import chainer as ch
v = ch.Variable(chx.ones(3, dtype='f'), requires_grad=True)
g = ch.Variable(chx.ones(3, dtype='f'), requires_grad=False)
v.grad_var = g
v.grad_var.requires_grad  # `True`, but should be `False`. It'll be `False` if `chx` is replaced with e.g. `numpy`.

```

Reply:
Fixed.